components/style/servo/media_queries.rs

```diff
@@ -252,9 +253,19 @@ fn eval_scan(_: &Context, _: Option<Scan>) -> bool {
     false
 }
 
+/// https://drafts.csswg.org/mediaqueries-4/#resolution
+fn eval_resolution(context: &Context) -> Resolution {
+    Resolution::from_dppx(context.device().device_pixel_ratio.0)
+}
+
+/// https://compat.spec.whatwg.org/#css-media-queries-webkit-device-pixel-ratio
+fn eval_device_pixel_ratio(context: &Context) -> f32 {
+    eval_resolution(context).dppx()
```

Comment:
Seems a bit strange to make `eval_resolution` wrap the `f32` into a `Resolution` just to immediately unwrap it. Might make more sense to swap this and make `eval_resolution` call `eval_device_pixel_ratio`:

```rs
/// https://drafts.csswg.org/mediaqueries-4/#resolution
fn eval_resolution(context: &Context) -> Resolution {
    Resolution::from_dppx(eval_device_pixel_ratio(context))
}

/// https://compat.spec.whatwg.org/#css-media-queries-webkit-device-pixel-ratio
fn eval_device_pixel_ratio(context: &Context) -> f32 {
    context.device().device_pixel_ratio.0
}
```

Reply:
I was thinking about this, but the way it's written now aligns better with spec:
> [-webkit-device-pixel-ratio](https://compat.spec.whatwg.org/#descdef-media--webkit-device-pixel-ratio) must be treated as an alias of the [resolution](https://drafts.csswg.org/mediaqueries-4/#descdef-media-resolution) range type [media feature](https://drafts.csswg.org/mediaqueries-4/#media-feature), with its value interpreted as a [dppx](https://drafts.csswg.org/css-values-3/#dppx) unit.

and mozilla's implemention: https://searchfox.org/mozilla-central/source/servo/components/style/gecko/media_features.rs#66